lib/http.c

```diff
@@ -2711,14 +2711,16 @@ CURLcode Curl_http_cookies(struct Curl_easy *data,
     int count = 0;
 
     if(data->cookies && data->state.cookie_engine) {
+      const char *host = data->state.aptr.cookiehost ?
+        data->state.aptr.cookiehost : conn->host.name;
+      const bool secure_context =
+        conn->handler->protocol&CURLPROTO_HTTPS ||
+        strcasecompare("localhost", host) ||
+        !strcmp(host, "127.0.0.1") ||
+        !strcmp(host, "[::1]") ? TRUE : FALSE;
```

Comment:
I'm torn, I sort of think this logic belong in `cookie.c` but only if the `secure` parameter instead takes the scheme.. so on the whole maybe this is the cleanest approach.

Reply:
Yeah, I'm not totally happy with this big conditional thing like this, and that it is mostly duplicated too, but I couldn't really figure out a better way to do it...

Reply:
I tried but failed as well,  so overall :+1: on this PR.